[main] Upgrade to latest dependencies

knative-automation opened the pull request:
Cron. -knative-prow-robot

/cc we don't use Prow
/assign we don't use Prow

Produced via:
```shell
./hack/update-deps.sh --upgrade --release v0.22
./hack/update-codegen.sh
```

Details:
```
bumping knative.dev/pkg c326b70...6a33a1a:
  > 6a33a1a Upgrade a few assorted dependencies (# 2058)
  > 4048853 Drop master references to knative repos (# 2050)
bumping knative.dev/eventing 9be1470...646b9e2:
  > 646b9e2 add unit tests to reconciler/apiserversource (# 5067)
  > b09cc03 add ./ to file (# 5078)
  > ef536d4 Start adding spec based conformance tests for Broker/Trigger. (# 5072)
  > 60043be Add KO_FLAGS usage to chaosduck build (# 5075)
  > b2ea385 Add redirect link (# 5077)
  > c8b0b6b Remove specs for eventing and point to new location in specs repo (# 5057)
  > 1184217 Update support info with machine-readable rotation (# 5068)
  > 248cd1b Add query for doc issues that eventing eng. attention (# 5074)
  > 2b95fce upgrade to latest dependencies (# 5060)
  > 4cbc070 Timezone not added to schedule in runner.go # 5058 (# 5059)
  > 3b42737 Smoke Tests on SinkBinding (# 5049)
  > 6cefb29 smoketest a channel (# 5052)
  > f80dc0c upgrade to latest dependencies (# 5056)
  > 52757a6 Format markdown (# 5055)
  > c8bfaea upgrade to latest dependencies (# 5054)
bumping github.com/google/go-cmp ec71d6d...8fa37b4:
  > 8fa37b4 Fix reporter verbosity bug (# 253)
  > dc6435e De-virtualize interfaces for specialized diffing (# 254)
  > e9947a2 Run tests on Go 1.16 (# 252)
  > 3a98a11 cmp/cmpopts: use errors.Is with ≥go1.13 in compareErrors (# 251)
```

n3wscott reviewed (dismissed):
/lgtm

Good job bot!

knative-automation committed:
upgrade to latest dependencies

bumping knative.dev/pkg c326b70...6a33a1a:
  > 6a33a1a Upgrade a few assorted dependencies (# 2058)
  > 4048853 Drop master references to knative repos (# 2050)
bumping knative.dev/eventing 9be1470...646b9e2:
  > 646b9e2 add unit tests to reconciler/apiserversource (# 5067)
  > b09cc03 add ./ to file (# 5078)
  > ef536d4 Start adding spec based conformance tests for Broker/Trigger. (# 5072)
  > 60043be Add KO_FLAGS usage to chaosduck build (# 5075)
  > b2ea385 Add redirect link (# 5077)
  > c8b0b6b Remove specs for eventing and point to new location in specs repo (# 5057)
  > 1184217 Update support info with machine-readable rotation (# 5068)
  > 248cd1b Add query for doc issues that eventing eng. attention (# 5074)
  > 2b95fce upgrade to latest dependencies (# 5060)
  > 4cbc070 Timezone not added to schedule in runner.go # 5058 (# 5059)
  > 3b42737 Smoke Tests on SinkBinding (# 5049)
  > 6cefb29 smoketest a channel (# 5052)
  > f80dc0c upgrade to latest dependencies (# 5056)
  > 52757a6 Format markdown (# 5055)
  > c8bfaea upgrade to latest dependencies (# 5054)
bumping github.com/google/go-cmp ec71d6d...8fa37b4:
  > 8fa37b4 Fix reporter verbosity bug (# 253)
  > dc6435e De-virtualize interfaces for specialized diffing (# 254)
  > e9947a2 Run tests on Go 1.16 (# 252)
  > 3a98a11 cmp/cmpopts: use errors.Is with ≥go1.13 in compareErrors (# 251)

Signed-off-by: Knative Automation <automation@knative.team>
